transport/http/outbound.go

```diff
@@ -399,6 +405,11 @@ func getYARPCErrorFromResponse(response *http.Response, bothResponseError bool)
 	)
 }
 
+func checkServiceMatch(reqSvcName string, resHeaders *http.Header) (bool, string) {
+	serviceName := resHeaders.Get(ServiceHeader)
+	return serviceName == "" || serviceName == reqSvcName, serviceName
```

Comment:
cc @abhinav is there an API for checking whether a header exists? is that better to use than the value?

Reply:
[http.Header](https://golang.org/pkg/net/http/#Header) doesn't have an API to check if something exists. Get returns `""` if there are no values associated with the provided name.

Reply:
It's just a `map` though, so we could check the map directly. It also doesn't have an API to get all the values for a specific key, I think users are expected to use the map operations directly.

Reply:
Thanks for the suggestion. `http.header` is a map of `map[string][]string`, which means we could have multiple values for a given key. `Get` method will wrap up canonical key mapping, return the first value in `[]string` or `""` if no value associated with that key. Perhaps the best practice is to only use raw map when we want to non-canonical keys or access multiple values of a key. Thus, use raw map could be tricky and it's safer to pair the usage of `Set()` and `Get()`.